diffkemp/building/cc_wrapper/cc_wrapper_utils.cpp

```diff
@@ -0,0 +1,98 @@
+#include <filesystem>
+#include <sstream>
+#include <string>
+#include <unistd.h>
+#include <unordered_map>
+#include <vector>
+
+std::vector<std::string> split(const std::string &s, char delim) {
+    std::vector<std::string> out;
+    std::stringstream ss(s);
+    std::string item;
+    while (std::getline(ss, item, delim)) {
+        out.push_back(item);
+    }
+    return out;
+}
+
+bool endsWith(const std::string &str, const std::string &suffix) {
+    return str.size() >= suffix.size()
+           && str.compare(str.size() - suffix.size(), suffix.size(), suffix)
+                      == 0;
+}
+
+// When making changes to this function, same changes need to be made
+// in diffkemp/llvm_ir/compiler.py
+std::vector<std::string> getClangDefaultOptions(bool defaultOptim = false) {
+    std::vector<std::string> options;
+    options = {
+            "-S", "-emit-llvm", "-g", "-fdebug-macro", "-Wno-format-security"};
+    if (defaultOptim) {
+        options.push_back("-O1");
+        options.push_back("-Xclang");
+        options.push_back("-disable-llvm-passes");
+    }
+    return options;
+}
+
+bool isExecutable(const std::string &path) {
+    return access(path.c_str(), X_OK) == 0;
+}
+
+std::string findInPath(const std::string &name) {
+    const char *envPath = std::getenv("PATH");
+    std::string pathstr = envPath ? envPath : std::string();
+    if (pathstr.empty()) {
+        pathstr = std::string(::getenv("PATH") ? ::getenv("PATH") : "");
+    }
+    std::stringstream ss(pathstr);
+    std::string dir;
+    while (std::getline(ss, dir, ':')) {
+        if (dir.empty()) {
+            continue;
+        }
+        std::string full = dir + "/" + name;
+        if (std::filesystem::exists(full) && isExecutable(full)) {
+            return full;
+        }
+    }
+    return "";
+}
+
+bool isInVector(const std::string &s, const std::vector<std::string> &vector) {
+    for (const std::string &val : vector) {
+        if (val == s) {
+            return true;
+        }
+    }
+    return false;
```

Comment:
This could be implemented by
```suggestion
    vector.find(s) != vector.end()
```
and probably inlined directly into the code.